xbmc/windows/GUIMediaWindow.cpp

```diff
@@ -900,7 +900,10 @@ bool CGUIMediaWindow::OnClick(int iItem, const std::string &player)
     else if (!CProfilesManager::GetInstance().GetCurrentProfile().canWriteSources() && !g_passwordManager.IsProfileLockUnlocked())
       return false;
 
-    return OnAddMediaSource();
+    if (OnAddMediaSource())
+      Refresh();
+
+    return true;
```

Comment:
@FernetMenta: it is. It's a list based cache. Whenever a call to `CDirectory::GetDirectory()` takes longer than X milliseconds (where X is 2000 or something) the whole result is cached based on it's path. All (well only the ones to "cacheable paths) calls to `CDirectory::GetDirectory()` check whether there's a cached list for the path being retrieved. If there is, the whole list is loaded from the cache. Whenever an item in the database is added, removed or changed the whole video/music db cache needs to be cleared because we don't know which item is cached for which paths.
So you need to call `Refresh(true)` here.

I've always hated this caching because it's a huge PITA especially when debugging with breakpoints (in which case the limit of X milliseconds is easily reached) but never bothered to replace it with something better.